OSS-Fuzz uses outdated Git submodule when triggered via Github Actions

Tarantool project has integration with OSS Fuzz via GH Actions workflow.

Pull request [#6731](https://github.com/tarantool/tarantool/pull/6731) contains a commit with update of one of the Git submodules. See commit with commit header "datetime: huge dates support in parse functions":
```
$ git show 833d35e4a5eaff4bc08d84642565e150ea29c5bc third_party/c-dt/
...
diff --git a/third_party/c-dt b/third_party/c-dt
index cbb3fc27c..cec6acebb 160000
--- a/third_party/c-dt
+++ b/third_party/c-dt
@@ -1 +1 @@
-Subproject commit cbb3fc27c104aa7703b01a4108ce7871e1a28a1c
+Subproject commit cec6acebb54d9e73ea0b99c63898732abd7683a6
```

GH Actions job is failed to check out a proper version of submodule `c-dt` ([log](https://github.com/tarantool/tarantool/runs/5756635410?check_suite_focus=true#step:4:18103), also attached to issue)
```
...
Run google/oss-fuzz/infra/cifuzz/actions/build_fuzzers@master
/usr/bin/docker run --name a[9](https://github.com/tarantool/tarantool/runs/5756635410?check_suite_focus=true#step:4:9)e75eed85f745247689329fa9aef5ef7ee_eeefde --label 372a9e --workdir /github/workspace --rm -e INPUT_OSS-FUZZ-PROJECT-NAME -e INPUT_DRY-RUN -e INPUT_SANITIZER -e INPUT_LANGUAGE -e INPUT_ALLOWED-BROKEN-TARGETS-PERCENTAGE -e INPUT_PROJECT-SRC-PATH -e INPUT_BAD-BUILD-CHECK -e INPUT_KEEP-UNAFFECTED-FUZZ-TARGETS -e OSS_FUZZ_PROJECT_NAME -e LANGUAGE -e DRY_RUN -e ALLOWED_BROKEN_TARGETS_PERCENTAGE -e SANITIZER -e PROJECT_SRC_PATH -e LOW_DISK_SPACE -e BAD_BUILD_CHECK -e CIFUZZ_DEBUG -e CFL_PLATFORM -e KEEP_UNAFFECTED_FUZZ_TARGETS -e HOME -e GITHUB_JOB -e GITHUB_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_REPOSITORY_OWNER -e GITHUB_RUN_ID -e GITHUB_RUN_NUMBER -e GITHUB_RETENTION_DAYS -e GITHUB_RUN_ATTEMPT -e GITHUB_ACTOR -e GITHUB_WORKFLOW -e GITHUB_HEAD_REF -e GITHUB_BASE_REF -e GITHUB_EVENT_NAME -e GITHUB_SERVER_URL -e GITHUB_API_URL -e GITHUB_GRAPHQL_URL -e GITHUB_REF_NAME -e GITHUB_REF_PROTECTED -e GITHUB_REF_TYPE -e GITHUB_WORKSPACE -e GITHUB_ACTION -e GITHUB_EVENT_PATH -e GITHUB_ACTION_REPOSITORY -e GITHUB_ACTION_REF -e GITHUB_PATH -e GITHUB_ENV -e GITHUB_STEP_SUMMARY -e RUNNER_OS -e RUNNER_ARCH -e RUNNER_NAME -e RUNNER_TOOL_CACHE -e RUNNER_TEMP -e RUNNER_WORKSPACE -e ACTIONS_RUNTIME_URL -e ACTIONS_RUNTIME_TOKEN -e ACTIONS_CACHE_URL -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/_temp/_runner_file_commands":"/github/file_commands" -v "/home/runner/work/tarantool/tarantool":"/github/workspace" 372a9e:75eed85f745247689329fa9aef5ef7ee
2022-03-30 15:[14](https://github.com/tarantool/tarantool/runs/5756635410?check_suite_focus=true#step:4:14):44,859 - root - DEBUG - Is github: True.
2022-03-30 [15](https://github.com/tarantool/tarantool/runs/5756635410?check_suite_focus=true#step:4:15):14:44,862 - root - DEBUG - base_commit: None
2022-03-30 15:14:44,862 - root - DEBUG - pr_ref: refs/pull/6731/merge
2022-03-30 15:14:44,862 - root - DEBUG - No PROJECT_SRC_PATH.
2022-03-30 15:14:44,863 - root - INFO - ci_system: <continuous_integration.InternalGithub object at 0x7f40f301faf0>.
2022-03-30 15:14:44,863 - root - INFO - ClusterFuzzDeployment: <clusterfuzz_deployment.OSSFuzz object at 0x7f40f301fc70>.
2022-03-30 15:14:44,863 - root - INFO - InternalGithub: preparing for fuzzer build.
2022-03-30 15:14:44,863 - root - INFO - Running: docker build -t gcr.io/oss-fuzz/tarantool --file /opt/oss-fuzz/projects/tarantool/Dockerfile /opt/oss-fuzz/projects/tarantool.
Sending build context to Docker daemon   7.68kB
...
Submodule path 'third_party/c-ares': checked out 'bbbffa4da8baf35b0e4e1c376e38018f9a8bcb4e'
Submodule path 'third_party/c-dt': checked out 'cbb3fc27c104aa7703b01a4108ce7871e1a28a1c'
Submodule path 'third_party/curl': checked out '3266b35bbe21c68dea0dc7ccd991eb028e6d360c'
Submodule path 'third_party/decNumber': checked out '5ddfde194cd68f7d8b9763d51c04b51d782df185'
...
[ 87%] Building C object src/lib/core/CMakeFiles/core.dir/iostream.c.o
cd /src/tarantool/build/src/lib/core && /usr/local/bin/clang -DCORO_ASM -DDT_NAMESPACE=tnt_ -DLUAJIT_SMART_STRINGS=1 -DNVALGRIND=1 -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS=1 -D__STDC_FORMAT_MACROS=1 -D__STDC_LIMIT_MACROS=1 -I/src/tarantool/src -I/src/tarantool/build/src -I/src/tarantool/src/lib -I/src/tarantool/src/lib/small/include -I/src/tarantool/src/lib/small/third_party -I/src/tarantool/src/lib/core -I/src/tarantool/third_party -I/src/tarantool/third_party/zstd/lib -I/src/tarantool/third_party/zstd/lib/common -I/src/tarantool/third_party/xxHash -I/src/tarantool/third_party/luajit/src -I/src/tarantool/build/third_party -I/src/tarantool/third_party/coro -I/src/tarantool/build/third_party/luajit/src -I/src/tarantool/third_party/libyaml/include -I/src/tarantool/src/lib/msgpuck -I/src/tarantool/build/build/curl/dest/include -I/src/tarantool/build/third_party/decNumber -I/src/tarantool/third_party/libutil_freebsd -I/src/tarantool/third_party/c-dt -O1 -fno-omit-frame-pointer -gline-tables-only -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION -fsanitize=array-bounds,bool,builtin,enum,float-divide-by-zero,function,integer-divide-by-zero,null,object-size,return,returns-nonnull-attribute,shift,signed-integer-overflow,unsigned-integer-overflow,unreachable,vla-bound,vptr -fno-sanitize-recover=array-bounds,bool,builtin,enum,float-divide-by-zero,function,integer-divide-by-zero,null,object-size,return,returns-nonnull-attribute,shift,signed-integer-overflow,unreachable,vla-bound,vptr -fsanitize=fuzzer-no-link -Wno-unknown-pragmas -fexceptions -funwind-tables -fno-common -msse2 -std=c11 -Wall -Wextra -Wno-strict-aliasing -fsanitize=alignment,bool,bounds,builtin,enum,float-cast-overflow,float-divide-by-zero,function,integer-divide-by-zero,return,shift,unreachable,vla-bound -fno-sanitize-recover=alignment,bool,bounds,builtin,enum,float-cast-overflow,float-divide-by-zero,function,integer-divide-by-zero,return,shift,unreachable,vla-bound -Wno-char-subscripts -Wno-gnu-alignof-expression -Wno-cast-function-type -Werror -g -ggdb -O0 -MD -MT src/lib/core/CMakeFiles/core.dir/iostream.c.o -MF CMakeFiles/core.dir/iostream.c.o.d -o CMakeFiles/core.dir/iostream.c.o -c /src/tarantool/src/lib/core/iostream.c
clang-14: warning: the object size sanitizer has no effect at -O0, but is explicitly enabled: -fsanitize=array-bounds,bool,builtin,enum,float-divide-by-zero,function,integer-divide-by-zero,null,object-size,return,returns-nonnull-attribute,shift,signed-integer-overflow,unsigned-integer-overflow,unreachable,vla-bound,vptr [-Winvalid-command-line-argument]
/src/tarantool/src/lib/core/datetime.c:100:7: error: implicit declaration of function 'dt_from_ymd_checked' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                if (dt_from_ymd_checked(year + 1900, mon + 1, mday, &dt) == false)
                    ^
1 error generated.
make[3]: *** [src/lib/core/CMakeFiles/core.dir/build.make:510: src/lib/core/CMakeFiles/core.dir/datetime.c.o] Error 1
make[3]: *** Waiting for unfinished jobs....
make[3]: Leaving directory '/src/tarantool/build'
```

Pay attention to commit hash used in submodule c-dt - `cbb3fc27c104aa7703b01a4108ce7871e1a28a1c`, but PR updates it to `cec6acebb54d9e73ea0b99c63898732abd7683a6`.

GH Action workflow log: [log.txt](https://github.com/google/oss-fuzz/files/8397789/log.txt)

### References:

1. Tarantool's fork with branch used in PR https://github.com/tsafin/tarantool/tree/tsafin/gh-5941-datetime-v6-3

--- a/infra/repo_manager.py
+++ b/infra/repo_manager.py
@@ -202,6 +202,8 @@ class RepoManager:
     self.fetch_unshallow()
     self.git(['fetch', 'origin', pr_ref], check_result=True)
     self.git(['checkout', '-f', 'FETCH_HEAD'], check_result=True)
+    self.git(['submodule', 'update', '-f', '--init', '--recursive'],
+             check_result=True)
 
   def checkout_commit(self, commit, clean=True):
     """Checks out a specific commit from the repo.
@@ -217,6 +219,8 @@ class RepoManager:
     if not self.commit_exists(commit):
       raise ValueError('Commit %s does not exist in current branch' % commit)
     self.git(['checkout', '-f', commit], check_result=True)
+    self.git(['submodule', 'update', '-f', '--init', '--recursive'],
+             check_result=True)
     if clean:
       self.git(['clean', '-fxd'], check_result=True)
     if self.get_current_commit() != commit:

